src/app/shared/components/TopBar/BreadCrumb/index.tsx

```diff
@@ -0,0 +1,49 @@
+'use client'
+
+import Image from "next/image"
+import Link from "next/link"
+
+import { getBreadCrumbPath } from '@functions/paths'
+
+import * as logos from '@logos/index'
+
+interface BreadCrumbProps {
+  entityKey: string,
+  isLast: boolean,
+  pathArray: string[],
+}
+
+const BreadCrumb = (props: BreadCrumbProps) => {
+  const { entityKey, isLast, pathArray } = props
+
+  const breadCrumbPath = getBreadCrumbPath(entityKey, pathArray)
```

Comment:
works, but i would do it a little differently, i would pass the `index` to `BreadCrumb` and then to `getBreadCrumbPath`

then we would already have the index in that function, we would not need to pass the `entityKey` or use `.findIndex`

Reply:
yep much cleaner